### PR TITLE
feat: Make command destructuring consistent.

### PR DESCRIPTION
**Pull request by WiseDevHelper**
**Please describe the changes this PR makes and why it should be merged:**

[Demo](https://deploy-preview-800--discordjs-guide.netlify.app/)

This PR updates all the sections of the guide to be consistent when destructuring `interaction.commandName` and adds destructuring where not present. It also updates some obsolete parts where necessary.

**Status**
- [x] Creating your bot.
  - Getting your bot up and running.¹
  - Configuration files.¹
  - [x] Adding more commands.
- [x] Command Handling.
- [x] Interactions.
  - Registering...¹
  - [x] Replying...
  - Permissions.¹
  - [x] Buttons.
  - [x] Select Menus.
- [x] Popular topics.
  - FAQ.¹
  - Threads.¹
  - Embeds.¹
  - Builders.¹
  - [x] Reactions.
  - Collectors.¹
  - [x] Permissions.²
  - Permissions (extended).
  - Gateway Intents.¹
  - Webhooks.¹
  - Errors.¹
  - Audit logs.¹
  - [x] Canvas.
- [x] Databases.
  - [x] Sequelize (Tags).
  - [x] Sequelize (Economy).  
  - KeyV¹
- [x] Sharding
  - [x] Getting started.
  - Additional info.
  - [x] Extended changes.
- [x] Additional Information.
  - Notation.¹
  - [x] ES6 Syntax.
  - Collections.¹
  - [x] Async/Await.
  - [x] REST API.
  - Updating to v13.³

**Reference**
¹ Does not require updating.
² Only requires updating of code samples.
³ Unchanged.

Please let me know if there is more to change. I'll be happy to work on them.

### CODE REVIEW

File: code-samples/popular-topics/canvas/13/index.js [hunk 1]
```diff
@@ -21,7 +21,9 @@ const applyText = (canvas, text) => {
 client.on('interactionCreate', async interaction => {
 	if (!interaction.isCommand()) return;
 
-	if (interaction.commandName === 'profile') {
+	const { commandName } = interaction;
+
+	if (commandName === 'profile') {
```
Comment (Danktuary) [inline code-samples/popular-topics/canvas/13/index.js:26]:
```suggestion
	if (interaction.commandName === 'profile') {
```

File: code-samples/popular-topics/reactions/13/awaiting-reactions.js [hunk 2]
```diff
@@ -11,7 +11,9 @@ client.once('ready', () => {
 client.on('interactionCreate', async interaction => {
 	if (!interaction.isCommand()) return;
 
-	if (interaction.commandName === 'react-await') {
+	const { commandName } = interaction;
+
+	if (commandName === 'react-await') {
```
Comment (Danktuary) [inline code-samples/popular-topics/reactions/13/awaiting-reactions.js:16]:
```suggestion
	if (interaction.commandName === 'react-await') {
```

File: guide/additional-info/async-await.md [hunk 3]
```diff
@@ -85,7 +85,11 @@ client.once('ready', () => {
 });
 
 client.on('interactionCreate', interaction => {
-	if (interaction.commandName === 'react') {
+	if (!interaction.isCommand()) return;
+
+	const { commandName } = interaction;
+
+	if (commandName === 'react') {
```
Comment (Danktuary) [inline guide/additional-info/async-await.md:92]:
```suggestion
	if (interaction.commandName === 'react') {
```

File: guide/interactions/buttons.md [hunk 4]
```diff
@@ -16,13 +16,15 @@ You can have a maximum of five `ActionRow`s per message, and five buttons within
 
 To create a button, use the `MessageActionRow()` and `MessageButton()` builder functions and then pass the resulting object to `CommandInteraction#reply()` as `InteractionReplyOptions`:
 
-```js {1,7-13,15}
+```js {1,9-15,17}
```
Comment (Danktuary) [inline guide/interactions/buttons.md:19]:
```suggestion
```js {1,7-13,15}
```

File: guide/interactions/buttons.md [hunk 5]
```diff
@@ -16,13 +16,15 @@ You can have a maximum of five `ActionRow`s per message, and five buttons within
...
 const { MessageActionRow, MessageButton } = require('discord.js');
 
 client.on('interactionCreate', async interaction => {
 	if (!interaction.isCommand()) return;
 
-	if (interaction.commandName === 'ping') {
+	const { commandName } = interaction;
+
+	if (commandName === 'ping') {
```
Comment (Danktuary) [inline guide/interactions/buttons.md:27]:
```suggestion
	if (interaction.commandName === 'ping') {
```

File: guide/interactions/buttons.md [hunk 6]
```diff
@@ -58,13 +60,15 @@ Restart your bot and then send the command to a channel your bot has access to.
 
 You can also send message components within an ephemeral response or alongside message embeds.
 
-```js {1,12-16,18}
+```js {1,14-18,20}
```
Comment (Danktuary) [inline guide/interactions/buttons.md:63]:
```suggestion
```js {1,12-16,18}
```

File: guide/interactions/buttons.md [hunk 7]
```diff
@@ -58,13 +60,15 @@ Restart your bot and then send the command to a channel your bot has access to.
...
 const { MessageActionRow, MessageButton, MessageEmbed } = require('discord.js');
 
 client.on('interactionCreate', async interaction => {
 	if (!interaction.isCommand()) return;
 
-	if (interaction.commandName === 'ping') {
+	const { commandName } = interaction;
+
+	if (commandName === 'ping') {
```
Comment (Danktuary) [inline guide/interactions/buttons.md:71]:
```suggestion
	if (interaction.commandName === 'ping') {
```

File: guide/interactions/replying-to-slash-commands.md [hunk 8]
```diff
@@ -34,11 +34,13 @@ The most common way of sending a response is by using the `CommandInteraction#re
 Initially an interaction token is only valid for three seconds, so that's the timeframe in which you are able to use the `CommandInteraction#reply()` method. Responses that require more time ("Deferred Responses") are explained later in this page.
 :::
 
-```js {1,4-6}
+```js {1,6-8}
```
Comment (Danktuary) [inline guide/interactions/replying-to-slash-commands.md:37]:
```suggestion
```js {1,4-6}
```

File: guide/interactions/replying-to-slash-commands.md [hunk 9]
```diff
@@ -34,11 +34,13 @@ The most common way of sending a response is by using the `CommandInteraction#re
...
 client.on('interactionCreate', async interaction => {
 	if (!interaction.isCommand()) return;
 
-	if (interaction.commandName === 'ping') {
+	const { commandName } = interaction;
+
+	if (commandName === 'ping') {
```
Comment (Danktuary) [inline guide/interactions/replying-to-slash-commands.md:43]:
```suggestion
	if (interaction.commandName === 'ping') {
```

File: guide/interactions/replying-to-slash-commands.md [hunk 10]
```diff
@@ -62,11 +64,13 @@ You've successfully sent a response to a slash command! This is only the beginni
 
 You may not always want everyone who has access to the channel to see a slash command's response. Thankfully, Discord implemented a way to hide messages from everyone but the executor of the slash command. This type of message is called `ephemeral` and can be set by using `ephemeral: true` in the `InteractionReplyOptions`, as follows:
 
-```js {5}
+```js {7}
```
Comment (Danktuary) [inline guide/interactions/replying-to-slash-commands.md:67]:
```suggestion
```js {5}
```

File: guide/interactions/replying-to-slash-commands.md [hunk 11]
```diff
@@ -62,11 +64,13 @@ You've successfully sent a response to a slash command! This is only the beginni
...
 client.on('interactionCreate', async interaction => {
 	if (!interaction.isCommand()) return;
 
-	if (interaction.commandName === 'ping') {
+	const { commandName } = interaction;
+
+	if (commandName === 'ping') {
```
Comment (Danktuary) [inline guide/interactions/replying-to-slash-commands.md:73]:
```suggestion
	if (interaction.commandName === 'ping') {
```

File: guide/interactions/replying-to-slash-commands.md [hunk 12]
```diff
@@ -95,13 +99,15 @@ After you've sent an initial response, you may want to edit that response for va
 After the initial response, an interaction token is valid for 15 minutes, so this is the timeframe in which you can edit the response and send follow-up messages.
 :::
 
-```js {1,8-9}
+```js {1,10-11}
```
Comment (Danktuary) [inline guide/interactions/replying-to-slash-commands.md:102]:
```suggestion
```js {1,8-9}
```

File: guide/interactions/replying-to-slash-commands.md [hunk 13]
```diff
@@ -95,13 +99,15 @@ After you've sent an initial response, you may want to edit that response for va
...
 const wait = require('util').promisify(setTimeout);
 
 client.on('interactionCreate', async interaction => {
 	if (!interaction.isCommand()) return;
 
-	if (interaction.commandName === 'ping') {
+	const { commandName } = interaction;
+
+	if (commandName === 'ping') {
```
Comment (Danktuary) [inline guide/interactions/replying-to-slash-commands.md:110]:
```suggestion
	if (interaction.commandName === 'ping') {
```

File: guide/interactions/replying-to-slash-commands.md [hunk 14]
```diff
@@ -116,13 +122,15 @@ As previously mentioned, you have three seconds to respond to an interaction bef
 In this case, you can make use of the `CommandInteraction#deferReply()` method, which triggers the `<application> is thinking...` message and also acts as initial response. This allows you 15 minutes to complete your tasks before responding.
 <!--- here either display the is thinking message via vue-discord-message or place a screenshot -->
 
-```js {7-9}
+```js {9-11}
```
Comment (Danktuary) [inline guide/interactions/replying-to-slash-commands.md:125]:
```suggestion
```js {7-9}
```

File: guide/interactions/replying-to-slash-commands.md [hunk 15]
```diff
@@ -116,13 +122,15 @@ As previously mentioned, you have three seconds to respond to an interaction bef
...
 const wait = require('util').promisify(setTimeout);
 
 client.on('interactionCreate', async interaction => {
 	if (!interaction.isCommand()) return;
 
-	if (interaction.commandName === 'ping') {
+	const { commandName } = interaction;
+
+	if (commandName === 'ping') {
```
Comment (Danktuary) [inline guide/interactions/replying-to-slash-commands.md:133]:
```suggestion
	if (interaction.commandName === 'ping') {
```

File: guide/interactions/replying-to-slash-commands.md [hunk 16]
```diff
@@ -148,10 +156,12 @@ Replying to slash commands is great and all, but what if you want to send multip
 After the initial response, an interaction token is valid for 15 minutes, so this is the timeframe in which you can edit the response and send follow-up messages.
 :::
 
-```js {6}
+```js {8}
```
Comment (Danktuary) [inline guide/interactions/replying-to-slash-commands.md:159]:
```suggestion
```js {6}
```

File: guide/interactions/replying-to-slash-commands.md [hunk 17]
```diff
@@ -148,10 +156,12 @@ Replying to slash commands is great and all, but what if you want to send multip
...
 client.on('interactionCreate', async interaction => {
 	if (!interaction.isCommand()) return;
 
+	const { commandName } = interaction;
+
```
Comment (Danktuary) [inline guide/interactions/replying-to-slash-commands.md:164]:
```suggestion
```

File: guide/interactions/replying-to-slash-commands.md [hunk 18]
```diff
@@ -313,10 +323,12 @@ const data = {
 
 The following snippet details the logic needed to parse the subcommands and respond accordingly using the `CommandInteractionOptionResolver#getSubcommand()` method:
 
-```js {5-20}
+```js {7-17}
 client.on('interactionCreate', async interaction => {
 	if (!interaction.isCommand()) return;
 
+	const { commandName } = interaction;
+
```
Comment (Danktuary) [inline guide/interactions/replying-to-slash-commands.md:331]:
```suggestion
```

File: guide/interactions/replying-to-slash-commands.md [hunk 19]
```diff
@@ -313,10 +323,12 @@ const data = {
 
 The following snippet details the logic needed to parse the subcommands and respond accordingly using the `CommandInteractionOptionResolver#getSubcommand()` method:
 
-```js {5-20}
+```js {7-17}
```
Comment (Danktuary) [inline guide/interactions/replying-to-slash-commands.md:326]:
```suggestion
```js {5-20}
```

File: guide/interactions/select-menus.md [hunk 20]
```diff
@@ -16,13 +16,15 @@ You can have a maximum of five `ActionRow`s per message, and one select menu wit
...
+```js {1,9-26,28}
 const { MessageActionRow, MessageSelectMenu } = require('discord.js');
 
 client.on('interactionCreate', async interaction => {
 	if (!interaction.isCommand()) return;
 
-	if (interaction.commandName === 'ping') {
+	const { commandName } = interaction;
+
+	if (commandName === 'ping') {
```
Comment (Danktuary) [inline guide/interactions/select-menus.md:27]:
```suggestion
	if (interaction.commandName === 'ping') {
```

File: guide/interactions/select-menus.md [hunk 21]
```diff
@@ -16,13 +16,15 @@ You can have a maximum of five `ActionRow`s per message, and one select menu wit
 
 To create a select menu, use the `MessageActionRow()` and `MessageSelectMenu()` builder functions and then pass the resulting object to `CommandInteraction#reply()` as `InteractionReplyOptions`:
 
-```js {1,7-24,26}
+```js {1,9-26,28}
```
Comment (Danktuary) [inline guide/interactions/select-menus.md:19]:
```suggestion
```js {1,7-24,26}
```

File: guide/interactions/select-menus.md [hunk 22]
```diff
@@ -67,13 +69,15 @@ Restart your bot and then send the command to a channel your bot has access to.
 
 You can also send message components within an ephemeral response or alongside message embeds.
 
-```js {1,12-16,18}
+```js {1,14-18,20}
```
Comment (Danktuary) [inline guide/interactions/select-menus.md:72]:
```suggestion
```js {1,12-16,18}
```

File: guide/interactions/select-menus.md [hunk 23]
```diff
@@ -67,13 +69,15 @@ Restart your bot and then send the command to a channel your bot has access to.
...
 const { MessageActionRow, MessageEmbed, MessageSelectMenu } = require('discord.js');
 
 client.on('interactionCreate', async interaction => {
 	if (!interaction.isCommand()) return;
 
-	if (interaction.commandName === 'ping') {
+	const { commandName } = interaction;
+
+	if (commandName === 'ping') {
```
Comment (Danktuary) [inline guide/interactions/select-menus.md:80]:
```suggestion
	if (interaction.commandName === 'ping') {
```

File: guide/interactions/select-menus.md [hunk 24]
```diff
@@ -154,11 +158,13 @@ The `MessageComponentInteraction` class provides the same methods as the `Comman
 
 The `MessageComponentInteraction` class provides an `update()` method to update the message the select menu is attached to. Passing an empty array to the `components` option will remove any menus after an option has been selected.
 
-```js {1,4-6}
+```js {1,6-8}
```
Comment (Danktuary) [inline guide/interactions/select-menus.md:161]:
```suggestion
```js {1,4-6}
```

File: guide/interactions/select-menus.md [hunk 25]
```diff
@@ -154,11 +158,13 @@ The `MessageComponentInteraction` class provides the same methods as the `Comman
...
 client.on('interactionCreate', async interaction => {
 	if (!interaction.isSelectMenu()) return;
 
-	if (interaction.customId === 'select') {
+	const { customId } = interaction;
+
+	if (customId === 'select') {
```
Comment (Danktuary) [inline guide/interactions/select-menus.md:167]:
```suggestion
	if (interaction.customId === 'select') {
```

File: guide/interactions/select-menus.md [hunk 26]
```diff
@@ -168,13 +174,15 @@ client.on('interactionCreate', async interaction => {
 
 Additionally to deferring the response of the interaction, you can defer the menu, which will trigger a loading state and then revert back to its original state:
 
-```js {1,6-10}
+```js {1,8-12}
```
Comment (Danktuary) [inline guide/interactions/select-menus.md:177]:
```suggestion
```js {1,6-10}
```

File: guide/interactions/select-menus.md [hunk 27]
```diff
@@ -168,13 +174,15 @@ client.on('interactionCreate', async interaction => {
...
 const wait = require('util').promisify(setTimeout);
 
 client.on('interactionCreate', async interaction => {
 	if (!interaction.isSelectMenu()) return;
 
-	if (interaction.customId === 'select') {
+	const { customId } = interaction;
+
+	if (customId === 'select') {
```
Comment (Danktuary) [inline guide/interactions/select-menus.md:185]:
```suggestion
	if (interaction.customId === 'select') {
```

File: guide/interactions/select-menus.md [hunk 28]
```diff
@@ -186,13 +194,15 @@ client.on('interactionCreate', async interaction => {
 
 A select menu is not bound to only one selection; you can specify a minimum and maximum amount of options that must be selected. You can use `MessageSelectMenu#setMinValues()` and `MessageSelectMenu#setMaxValues()` to determine these values.
 
-```js {1,7-31,33}
+```js {1,9-33,35}
```
Comment (Danktuary) [inline guide/interactions/select-menus.md:197]:
```suggestion
```js {1,7-31,33}
```

File: guide/interactions/select-menus.md [hunk 29]
```diff
@@ -186,13 +194,15 @@ client.on('interactionCreate', async interaction => {
...
 const { MessageActionRow, MessageSelectMenu } = require('discord.js');
 
 client.on('interactionCreate', async interaction => {
 	if (!interaction.isCommand()) return;
 
-	if (interaction.commandName === 'ping') {
+	const { commandName } = interaction;
+
+	if (commandName === 'ping') {
```
Comment (Danktuary) [inline guide/interactions/select-menus.md:205]:
```suggestion
	if (interaction.commandName === 'ping') {
```

File: guide/popular-topics/canvas.md [hunk 30]
```diff
@@ -47,7 +47,9 @@ client.once('ready', () => {
 client.on('interactionCreate', interaction => {
 	if (!interaction.isCommand()) return;
 
-	if (interaction.commandName === 'profile') {
+	const { commandName } = interaction;
+
+	if (commandName === 'profile') {
```
Comment (Danktuary) [inline guide/popular-topics/canvas.md:52]:
```suggestion
	if (interaction.commandName === 'profile') {
```

File: guide/popular-topics/canvas.md [hunk 31]
```diff
@@ -69,11 +71,13 @@ After importing the Canvas module and initializing it, you should load the image
 `canvas` works almost identical to HTML5 Canvas. You can read the HTML5 Canvas tutorials on [w3Schools](https://www.w3schools.com/html/html5_canvas.asp) and [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Canvas_API) for more information later!
 :::
 
-```js {5-8}
+```js {7-10}
```
Comment (Danktuary) [inline guide/popular-topics/canvas.md:74]:
```suggestion
```js {5-8}
```

File: guide/popular-topics/canvas.md [hunk 32]
```diff
@@ -69,11 +71,13 @@ After importing the Canvas module and initializing it, you should load the image
...
 client.on('interactionCreate', async interaction => {
 	if (!interaction.isCommand()) return;
 
-	if (interaction.commandName === 'profile') {
+	const { commandName } = interaction;
+
+	if (commandName === 'profile') {
```
Comment (Danktuary) [inline guide/popular-topics/canvas.md:80]:
```suggestion
	if (interaction.commandName === 'profile') {
```

**Review by Danktuary — COMMENTED** [review]
Single access use doesn't need to be destructured

File: guide/interactions/replying-to-slash-commands.md [hunk 33]
```diff
@@ -148,10 +156,12 @@ Replying to slash commands is great and all, but what if you want to send multip
...
 :::
 
-```js {6}
+```js {8}
 client.on('interactionCreate', async interaction => {
 	if (!interaction.isCommand()) return;
 
+	const { commandName } = interaction;
+
 	if (interaction.commandName === 'ping') {
```
Comment (WiseDevHelper) [inline guide/interactions/replying-to-slash-commands.md:165]:
```suggestion
	if (interaction.commandName === 'ping') {
```

File: guide/interactions/replying-to-slash-commands.md (hunk 19 above)
Comment (WiseDevHelper) [reply guide/interactions/replying-to-slash-commands.md:326]:
```suggestion
```js {5-15}
```

File: code-samples/command-handling/file-setup/13/index.js [hunk 34]
```diff
@@ -18,11 +18,13 @@ client.once('ready', () => {
 
 client.on('interactionCreate', async interaction => {
 	if (!interaction.isCommand()) return;
-
-	if (!client.commands.has(interaction.commandName)) return;
+
```
Comment (WiseDevHelper) [inline code-samples/command-handling/file-setup/13/index.js:21]:
```suggestion

```

File: code-samples/command-handling/file-setup/13/index.js [hunk 35]
```diff
@@ -18,11 +18,13 @@ client.once('ready', () => {
 
 client.on('interactionCreate', async interaction => {
 	if (!interaction.isCommand()) return;
-
-	if (!client.commands.has(interaction.commandName)) return;
+	
+	const { commandName } = interaction;
+
```
Comment (WiseDevHelper) [inline code-samples/command-handling/file-setup/13/index.js:23]:
```suggestion

```

File: guide/interactions/replying-to-slash-commands.md [hunk 36]
```diff
@@ -313,10 +323,12 @@ const data = {
...
 The following snippet details the logic needed to parse the subcommands and respond accordingly using the `CommandInteractionOptionResolver#getSubcommand()` method:
 
-```js {5-20}
+```js {7-17}
 client.on('interactionCreate', async interaction => {
 	if (!interaction.isCommand()) return;
 
+	const { commandName } = interaction;
+
 	if (interaction.commandName === 'info') {
```
Comment (WiseDevHelper) [inline guide/interactions/replying-to-slash-commands.md:332]:
```suggestion
	if (interaction.commandName === 'info') {
```

**Pull request merged**